### PR TITLE
Fix [1277] for cross-thread exception using ownerWindow

### PR DESCRIPTION
**Pull request by MarkZuber**

### CODE REVIEW

**Review by henrik-me — APPROVED** [review]
:shipit:

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs [hunk 1]
```diff
@@ -351,11 +355,31 @@ protected virtual void OnAuthenticate()
...
 
+        /// <summary>
+        /// Some calls need to be made on the UI thread and this is the central place to check if we have an owner
+        /// window and if so, ensure we invoke on that proper thread.
+        /// </summary>
+        /// <param name="action"></param>
+        protected void InvokeHandlingOwnerWindow(Action action)
+        {
+            // We only support WindowsForms (since our dialog is winforms based)
+            if (ownerWindow != null && ownerWindow is Control winFormsControl)
```
Comment (jennyf19) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs:366]: >winFormsControl [](start = 62, length = 15)

cool

**Review by jennyf19 — APPROVED** [review]

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs [hunk 2]
```diff
@@ -351,11 +355,31 @@ protected virtual void OnAuthenticate()
...
         }
 
+        /// <summary>
+        /// Some calls need to be made on the UI thread and this is the central place to check if we have an owner
+        /// window and if so, ensure we invoke on that proper thread.
+        /// </summary>
+        /// <param name="action"></param>
+        protected void InvokeHandlingOwnerWindow(Action action)
+        {
+            // We only support WindowsForms (since our dialog is winforms based)
```
Comment (bgavrilMS) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs:365]: Will this work for a WPF application? i.e. is the ownerWindow a Control object?

Comment (MarkZuber) [reply adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs:365]: No, but our browser is only WinForms and the owner window can only be an IWin32Window or we fail.  We should investigate an option for embedding Edge which they now have as a WPF and WinForms plugin.

**Pull request merged**